DotNetDBF/DBFReader.cs

```diff
@@ -245,175 +244,156 @@ internal Object[] NextRecord(IEnumerable<int> selectIndexes, IList<int> sortedIn
+                                    {
+                                        recordObjects[i] = Convert.ToSingle(tParsed, NumberFormatProvider);
+                                    }
+                                    else
+                                    {
+                                        recordObjects[i] = null;
+                                    }
                                 }
-                                else
+                                catch (FormatException e)
                                 {
-                                    recordObjects[i] = null;
+                                    throw new DBFException("Failed to parse Float", e);
                                 }
-                            }
-                            catch (FormatException e)
-                            {
-                                throw new DBFException("Failed to parse Float",
-                                                       e);
-                            }
-
-                            break;
 
+                                break;
+                            }
                         case NativeDbType.Numeric:
-
-                            try
                             {
-                                byte[] t_numeric = new byte[
-                                    _header.FieldArray[i].FieldLength
-                                    ];
-                                _dataInputStream.Read(t_numeric,
-                                                     0,
-                                                     t_numeric.Length);
-                                string tParsed =
-                                    CharEncoding.GetString(t_numeric);
-                                var tLast = tParsed.Substring(tParsed.Length - 1);
-                                if (tParsed.Length > 0
-                                    && tLast != " "
-                                    && tLast != DBFFieldType.Unknown)
+                                try
                                 {
-                                    recordObjects[i] = Decimal.Parse(tParsed, NumberStyles.Float | NumberStyles.AllowLeadingWhite);
+                                    byte[] t_numeric = new byte[_header.FieldArray[i].FieldLength];
+                                    _dataInputStream.Read(t_numeric, 0, t_numeric.Length);
+                                    string cellValue = CharEncoding.GetString(t_numeric);
+                                    string lastChar = cellValue.Substring(cellValue.Length - 1);
+                                    if (cellValue.Length > 0
+                                        && lastChar != " "
+                                        && lastChar != DBFFieldType.Unknown)
+                                    {
+                                        recordObjects[i] = Convert.ToDouble(cellValue, NumberFormatProvider);

```

Comment:
This is the part I have changed. You see, I had a case when value in DBF table was "0.436". But because of default culture settings `double.Parse` expected decimal separator to be a comma (because it's a default behavior in my culture). I also renamed some of variables in this block f code (while I was trying to figure out how it works) this just to better understand their meaning.